scripts/tidycensus.py

```diff
@@ -0,0 +1,220 @@
+
+                df = pd.read_csv(path,
+                                 encoding='cp1252',
+                                 converters={'code': lambda x: str(x)})
+                df.to_csv(path, encoding='utf-8', index=False)
+                file_paths['mig_recodes'] = path
+
+            elif dataset_name in ['county_laea', 'state_laea']:
+                path = os.path.normpath(
+                    os.path.join(self.engine.format_data_dir(), str(dataset_name)))
+                r_string = '''
+                dir.create("{path}/")
+                setwd("{path}/")
+                library(tidycensus)
+                library(sf)
+                st_write({dataset}, "{dataset}.shp", delete_layer=TRUE)
+                '''.format(path=path, dataset=dataset_name)
+
+                ro.r(r_string)
+                file_paths[dataset_name] = os.path.normpath(
+                    os.path.join(path, '{}.shp'.format(dataset_name)))
+
+            else:
+                r_string = '''
+                library(tidycensus)
+                rdf = as.data.frame.data.frame({})
+                '''.format(dataset_name)
+
+                ro.r(r_string)
+                rdf = ro.r('rdf')
+
+                with localconverter(ro.default_converter + pandas2ri.converter):
+                    pdf = ro.conversion.rpy2py(rdf)
+                path = os.path.normpath(
+                    self.engine.format_filename('{}.csv'.format(dataset_name)))
+                pdf.to_csv(path, index=False)
+                file_paths[dataset_name] = path
+
+    def download(self, engine=None, debug=False):
+        Script.download(self, engine, debug)
+        engine = self.engine
+        datasets = [
+            'fips_codes', 'mig_recodes', 'pums_variables', 'state_laea', 'county_laea'
+        ]
+        key = ro.r('Sys.getenv("CENSUS_API_KEY")')
+
+        data_path = os.path.normpath(DATA_WRITE_PATH.format(dataset='tidycensus'))
+        if not os.path.exists(data_path):
+            engine.create_raw_data_dir()
+
+        download = True
+        filepaths = [
+            'fips_codes.csv', 'mig_recodes.csv', 'pums_variables.csv',
+            'state_laea/state_laea.shp', 'county_laea/county_laea.shp'
+        ]
+        for filepath in filepaths:
+            if os.path.isfile(os.path.join(data_path, filepath)):
+                filename, file_ext = os.path.splitext(os.path.basename(filepath))
+                file_paths[filename] = os.path.join(data_path, filepath)
+                download = False
+
+        if (key[0]
+                == '') and (len([x for x in packages if rpackages.isinstalled(x)]) > 0):
+            result = self.install_tidycensus()
+            print("Installing tidycensus...")
+            if result:
+                print("Installation completed!!!")
+                print(
+                    '*The API key provided to you from the Census formated in quotes. A key can be acquired at http://api.census.gov/data/key_signup.html*'
```

Comment:
Avoid very long lines, you have some tools that you could use to do some refactoring like yapf. But be care full of applying it to scripts. Especially the the very first lines should not be changed.
```python
# -*- coding: utf-8 -*-
#retriever
```